### PR TITLE
feat: `sequence` support `BatchMsg`

### LINKED ISSUE #582: feat: add support for `BatchMsg` in `sequenceMsg`
#415 added new `Sequence` command. This command should run the passed commands in order. However it doesn't take `Batch` command into account.

### PR DESCRIPTION
**Pull request by stoffeastrom**
#415 added support for `tea.Sequence`. This adds support for `tea.Batch` cmd in the `tea.Sequence` cmd.

Fixes #582

### CODE REVIEW

**Comment by meowgorithm** [comment]
This makes a ton of sense to me; thank you for the PR. We _just_ merged another PR that touches this portion this code and is now causing a conflict. Do you mind updating your PR accordingly?

**Comment by stoffeastrom** [comment]
@meowgorithm Rebased. Let me know if anything else is needed!

**Comment by stoffeastrom** [comment]
@meowgorithm saw there was some mismatch in `go.sum` so updated those as well.

**Comment by meowgorithm** [comment]
> @meowgorithm saw there was some mismatch in `go.sum` so updated those as well.

Thank you!

File: tea.go [hunk 1]
```diff
@@ -313,8 +314,22 @@ func (p *Program) eventLoop(model Model, cmds chan Cmd) (Model, error) {
 						if cmd == nil {
 							continue
 						}
-
-						p.Send(cmd())
+						msg := cmd()
+						switch msg := msg.(type) {
```
[the remark this answers is not in the document]
Comment (stoffeastrom) [reply tea.go]: sure, updated accordingly!

**Comment by stoffeastrom** [comment]
Apparently I missed updating `go.sum` in tutorials so fixed that as well

**Comment by meowgorithm** [comment]
Alright, this looks good! Thanks for all your help with this, @stoffeastrom!

**Pull request merged**